src/ixdat/readers/zilien.py

```diff
@@ -1,61 +1,314 @@
+        """
+        if self._path_to_file:
+            print(
+                f"This {self.__class__.__name__} has already read {self._path_to_file}. "
+                "Returning the measurement resulting from the original read. "
+                "Use a new Reader if you want to read another file."
+            )
+            return self._measurement
+
+        if "technique" in kwargs:
+            if kwargs["technique"] == "EC-MS":
+                cls = ECMSMeasurement
+            if kwargs["technique"] == "EC":
+                cls = ECMeasurement
+            if kwargs["technique"] == "MS":
+                cls = MSMeasurement
+        else:
+            if cls is Measurement:
+                cls = ECMSMeasurement
 
-        if "technique" not in kwargs:
             if issubclass(cls, ECMSMeasurement):
                 kwargs["technique"] = "EC-MS"
             elif issubclass(cls, ECMeasurement):
                 kwargs["technique"] = "EC"
             elif issubclass(cls, MSMeasurement):
                 kwargs["technique"] = "MS"
+        self._cls = cls
+
+        self._path_to_file = Path(path_to_file)
+
+        # Extra timestamp from filename on form: 2021-04-20 11_16_18 Measurement name
+        file_stem = (
+            self._path_to_file.stem
+        )  # The part of the filename before the extension
+        timestamp_string = " ".join(self._path_to_file.stem.split(" ")[:2])
+        timestamp = time.mktime(
+            datetime.datetime.strptime(
+                timestamp_string, ZILIEN_TIMESTAMP_FORM
+            ).timetuple()
+        )
+
+        # Parse metadata items
+        with open(self._path_to_file, encoding="utf-8") as file_handle:
+            metadata, series_headers, column_headers = self._read_metadata(file_handle)
+            file_position = file_handle.tell()
 
-        ec_ms_dataset = Zilien_Dataset(path_to_file)
+        # Read raw data
+        with open(self._path_to_file, "rb") as file_handle:
+            file_handle.seek(file_position)
+            data = np.genfromtxt(file_handle, delimiter="\t")
 
-        return measurement_from_ec_ms_dataset(
-            ec_ms_dataset.data,
-            cls=cls,
-            name=name,
-            reader=self,
-            aliases=ZILIEN_LEGACY_ALIASES,
-            **kwargs,
+        # Extract series data and form series
+        series, aliases = self._form_series(
+            data, metadata, timestamp, series_headers, column_headers
         )
+        for standard_name, general_aliases in ZILIEN_ALIASES[self._cls].items():
+            aliases[standard_name] += general_aliases
+        aliases = dict(aliases)  # Convert from defaultdict to normal dict
+
+        measurement_kwargs = {
+            "name": name or file_stem,
+            "series_list": series,
+            "aliases": aliases,
+            "tstamp": timestamp,
+            "metadata": metadata,
+        }
+        self._measurement = cls(**measurement_kwargs)
```

Comment:
Hmm. I agree that the update line is missing. Nice catch. I'm a little puzzled, because it hasn't failed for me. In any case, I have added a test case for each of the three options of read them from `Measurement` and with differing techniques.

Reply:
It worked when you import the class, as you do in the test. With this update it also work when the technique is passed as a kwarg.